woocommerce/woocommerce-gutenberg-products-block
# Fix wide appender buttons overlap in Cart & Checkout blocks in the Editor

Since this [Gutenberg change](https://github.com/WordPress/gutenberg/pull/36605) all appender buttons have  `position:absolute` resulting in layout changes for the wide appender button.

Initially I've tried to remove the renderAppender prop from places like [here ](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/446fade219c3f31058947d7560a203f834446dd4/assets/js/blocks/checkout/inner-blocks/checkout-totals-block/edit.tsx#L60) so that the default black Add block button should be shown instead of the white one with black borders. That worked, but the button was visible only when the block was selected. We needed the appender buttons to be visible all the time.

I've decided to fix this in place via CSS, so that we have the old buttons working as before.

I’m doing some research, and it looks like [the GB change ](https://github.com/WordPress/gutenberg/pull/36605)made people unhappy, as it ruined their layouts as well. I’ve also added a [comment](https://github.com/WordPress/gutenberg/pull/36605), I hope they will only make the small black buttons absolute and let the others as before so that we can revert this change.


Fixes #5742 




### Screenshots

| Before | After |
| ------------ | ------ |
| ![Screenshot 2022-02-08 at 11 16 59](https://user-images.githubusercontent.com/3966773/152976612-e3895c81-7a56-4aca-8762-bd8660a64f6a.png) | <img width="414" alt="Screenshot 2022-02-09 at 16 23 02" src="https://user-images.githubusercontent.com/1628454/153232357-a22edf4a-29f3-4c95-89dd-6a3443621cce.png"> |
|![Screenshot 2022-02-08 at 11 16 51](https://user-images.githubusercontent.com/3966773/152976700-e117efd1-b6c8-4292-bacd-976954e519a6.png) | <img width="642" alt="Screenshot 2022-02-09 at 16 23 37" src="https://user-images.githubusercontent.com/1628454/153232471-047fb440-7050-4fa1-8021-2d225cbc354f.png"> |


### Testing

### Manual Testing

How to test the changes in this Pull Request:

1. Open up a page with the Checkout block in the editor
2. Confirm that the "Place Order" button and the "Add Block" button at the bottom of the block don't overlap
3. Confirm that in the right hand side bar, the total price doesn't overlaps with the "Add block" button
4. Also check the Cart block that the insert inner block buttons don't overlap

### User Facing Testing
These are steps for user testing (where "user" is someone interacting with this change that is not editing any code).




### Performance Impact



### Changelog

> Fix wide appender buttons overlap in Cart & Checkout blocks in the Editor
